### ISSUE TITLE
Circular reference with ParamConverter annotation

### ISSUE THREAD

**Issue by slavafomin** [symfony/symfony#14337]
Hello!

I've encountered some weird issue when using [ParamConverter](http://symfony.com/doc/current/bundles/SensioFrameworkExtraBundle/annotations/converters.html#doctrine-converter) in my Symfony 2 application (`2.6.6`).

I have the following controller:

```php
    /**
     * @param User $user
     * @param Mailing $mailing
     *
     * @return array
     *
     * @REST\Get("/users/{id}/mailings/{mailingId}")
     * @REST\View
     *
     * @ParamConverter("mailing", options={"id" = "mailingId"})
     */
    public function getMailingForUserAction(User $user, Mailing $mailing)
    {
        return [];
    }
```

I'm getting the following exception:

```
FileLoaderImportCircularReferenceException in FileLoader.php line 97: Circular reference detected in "/vagrant/app/config/routing.yml" ("/vagrant/app/config/routing.yml" > "/vagrant/app/config/routing_api.yml" > "AppBundle\Controller\MailingsController" > "/vagrant/app/config/routing.yml").
```

The same problem [is discussed here](http://forum.symfony-project.org/viewtopic.php?p=123764#p123764), with no answer, so it's not the isolated case.

**Comment by javiereguiluz** [comment]
@slavafomin thanks for reporting this issue.

Apparently, the problem is not related to the ParamConverter. Lately we've received a lot of reports of misleading errors referring to *circular reference errors* when in fact the problem has nothing to do with that (see #14199 and #13963).

@nicolas-grekas worked on a solution for this problem and closed a related issue (#14073). But it looks like we may have to work a bit more on this issue.

**Comment by xabbuh** [comment]
closing in favor of #14342
